### PR TITLE
Azure pr testing job for 3.9

### PR DESCRIPTION
**Pull request by pschiffe**
@kwoodson @jim-minter

### CODE REVIEW

**Comment by pschiffe** [comment]
```
TASK [openshift_azure : Create storage account] ********************************
Monday 09 April 2018  20:24:37 +0000 (0:00:02.220)       0:00:02.297 ********** 

fatal: [localhost]: FAILED! => {"changed": false, "msg": "Parameter error: name length must be between 3 and 24 characters."}
```

File: sjb/config/common/test_cases/origin_release_install_azure.yml [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      default_value: "registry.svc.ci.openshift.org/"
+    - name: "OS_PUSH_BASE_REPO"
+      default_value: "ci-pr-images/"
+  actions:
+    - type: "forward_parameters"
+      parameters:
+        - OPENSHIFT_ANSIBLE_IMAGE
+        - ADDITIONAL_SKIP
+        - JOB_NAME
+        - BUILD_NUMBER
```
Comment (kargakis) [inline sjb/config/common/test_cases/origin_release_install_azure.yml]: I think you also need BUILD_ID

Comment (kargakis) [reply sjb/config/common/test_cases/origin_release_install_azure.yml]: Or maybe not

File: sjb/config/common/test_cases/origin_release_install_azure.yml [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+        suffix="${suffix}:$( cat /data/src/github.com/openshift/aos-cd-jobs/ORIGIN_TAG )"
+        pushd release/cluster/test-deploy/${CLUSTER_PROFILE}
+        TYPE=$CLUSTER_PROFILE ../../bin/ansible.sh ansible-playbook \
+          -e "openshift_test_repo=${location_url}" \
+          -e "debug_level=5" \
+          -e "openshift_azure_base_image_ns=opstest" \
+          -e "openshift_azure_base_image=rhel7.4_base-20180323133134" \
+          -e "openshift_azure_image=${INSTANCE_PREFIX}-openshift-gi" \
+          -e "openshift_azure_clusterid=magentaci${INSTANCE_PREFIX}" \
+          playbooks/${CLUSTER_PROFILE}/openshift-cluster/build_image.yml
```
Comment (kargakis) [inline sjb/config/common/test_cases/origin_release_install_azure.yml]: Is this the correct playbook for launching a cluster?

**Comment by kargakis** [comment]
Rename `test_branch_extended_conformance_azure.yml` to `test_branch_origin_extended_conformance_azure.yml`. Also, try to replicate the GCE jobs: you also need a `test_pull_request_origin_extended_conformance_azure.yml`. No difference into what these jobs will do but let's stay consistent.

File: sjb/config/common/test_cases/origin_release_install_azure.yml [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+  #   - type: "script"
+  #     title: "deprovision test cluster"
+  #     repository: "release"
+  #     timeout: 900
+  #     script: |-
+  #       trap 'exit 0' EXIT
+  #       pushd release/cluster/test-deploy/${CLUSTER_PROFILE}
+  #       TYPE=$CLUSTER_PROFILE ../../bin/ansible.sh ansible-playbook \
+  #         -e "openshift_azure_clusterid=magentaci${INSTANCE_PREFIX}" \
+  #         playbooks/${CLUSTER_PROFILE}/openshift-cluster/deprovision.yml
```
Comment (kargakis) [inline sjb/config/common/test_cases/origin_release_install_azure.yml]: Seems like @kwoodson's branch has this playbook around. We should do deprovisioning.

File: sjb/config/common/test_cases/origin_release_install_azure.yml [hunk 4]
```diff
@@ -0,0 +1,194 @@
+---
+parent: 'common/test_cases/origin_release_with_ecosystem.yml'
```
Comment (kargakis) [inline sjb/config/common/test_cases/origin_release_install_azure.yml]: Can we run our job against openshift/origin#master today? Shouldn't we test release-3.9 instead (in which case this should be `common/test_cases/origin_release_with_ecosystem_39.yml`)?

**Comment by pschiffe** [comment]
@kargakis thanks for the reviews, I've updated the code now.

**Pull request merged**